### PR TITLE
Slight refactoring

### PR DESCRIPTION
**Pull request by nvuillam**
Please can you merge in your main branch, then create a sub branch, then check it's all ok for you, then create a new PR to the master repo ? :)

I've not tested my refactos so still a little bit of rework probably

Sorry for the mess ^^

### CODE REVIEW

**Comment by nvuillam** [comment]
@masaner I think it's ok now, you can merge then make a PR if it's ok for you too :)

File: src/commands/hardis/org/diagnose/unusedusers.ts [hunk 1]
```diff
@@ -179,43 +125,98 @@ export default class DiagnoseUnusedUsers extends SfdxCommand {
...
+      }
+    }
+  }
+
   private async listUnusedUsersWithSfdcLicense(conn) {
-    let whereConstraint = `WHERE IsActive = true AND LastLoginDate != LAST_N_DAYS:${this.lastNdays} AND LastLoginDate != NULL`;
-    
+    let whereConstraint = `WHERE IsActive = true AND (` +
+      `(LastLoginDate != LAST_N_DAYS:${this.lastNdays} AND LastLoginDate != NULL) OR ` +
+      `(CreatedDate != LAST_N_DAYS:${this.lastNdays} AND LastLoginDate == NULL)` + // Check also for users never used
```
Comment (masaner) [inline src/commands/hardis/org/diagnose/unusedusers.ts]: Should this query be updated? 

FROM: AND LastLoginDate == NULL
TO: AND LastLoginDate = NULL

**Review by masaner — CHANGES_REQUESTED** [review]

File: src/commands/hardis/org/diagnose/unusedusers.ts [hunk 2]
```diff
@@ -179,43 +135,97 @@ export default class DiagnoseUnusedUsers extends SfdxCommand {
...
+      }
+    }
+  }
+
   private async listUnusedUsersWithSfdcLicense(conn) {
-    let whereConstraint = `WHERE IsActive = true AND LastLoginDate != LAST_N_DAYS:${this.lastNdays} AND LastLoginDate != NULL`;
-    
+    let whereConstraint = `WHERE IsActive = true AND (` +
+      `(LastLoginDate < LAST_N_DAYS:${this.lastNdays} AND LastLoginDate != NULL) OR ` +
+      `(CreatedDate < LAST_N_DAYS:${this.lastNdays} AND LastLoginDate = NULL)` + // Check also for users never used
```
Comment (masaner) [inline src/commands/hardis/org/diagnose/unusedusers.ts:210]: Oh, nvm, i see its changed here.

**Review by masaner — APPROVED** [review]
Looks good!

**Pull request merged**